From f3772ad353f452c82560b72ed43ba07ad95cee76 Mon Sep 17 00:00:00 2001
From: Jiang Lu <lu.jiang@windriver.com>
Date: Mon, 6 May 2019 16:14:16 +0800
Subject: [PATCH 2/8] arm:dts:imx7d-sdb:move wdag & backlight pinctrls into
 imx7d-sdb

imx-pinctrl driver expect all child nodes with 'fsl,pins' property
defined in same level under iomuxc device node.

In imx7d-sdb.dts, it defined a imx7d-sdb node under iomuxc-lpsr device
to contain pinctrl device nodes.
The driver can not parse other pinctrl nodes in iomuxc-lpsr device, it
leads following error:

imx7d-pinctrl 302c0000.iomuxc-lpsr: no groups defined in /soc/aips-bus@30000000/iomuxc-lpsr@302c0000/wdoggrp
imx7d-pinctrl 302c0000.iomuxc-lpsr: no groups defined in /soc/aips-bus@30000000/iomuxc-lpsr@302c0000/backlightgrp

To fix, need move those pinctrl nodes into imx7d-sdb nodes in iomuxc-lpsr.

Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 arch/arm/boot/dts/imx7d-sdb.dts | 20 +++++++++++---------
 1 file changed, 11 insertions(+), 9 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index cb21b37f9284..20d809d1ae19 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -1228,15 +1228,17 @@
 };
 
 &iomuxc_lpsr {
-	pinctrl_wdog: wdoggrp {
-		fsl,pins = <
-			MX7D_PAD_LPSR_GPIO1_IO00__WDOG1_WDOG_B		0x74
-		>;
-	};
+	imx7d-sdb {
+		pinctrl_wdog: wdoggrp {
+			fsl,pins = <
+				MX7D_PAD_LPSR_GPIO1_IO00__WDOG1_WDOG_B		0x74
+			>;
+		};
 
-	pinctrl_backlight: backlightgrp {
-		fsl,pins = <
-			MX7D_PAD_LPSR_GPIO1_IO01__GPIO1_IO1		0x110b0
-		>;
+		pinctrl_backlight: backlightgrp {
+			fsl,pins = <
+				MX7D_PAD_LPSR_GPIO1_IO01__GPIO1_IO1		0x110b0
+			>;
+		};
 	};
 };
-- 
2.17.1

